### PR TITLE
feat: site discovery

### PR DESCRIPTION
**Pull request by ekremney**
This pull request (PR) implements automated site discovery from CDN and RUM logs. Incoming webhooks provide the consumer with the x-forwarded-host header or the RUM domain.

1. Inputs undergo sanitization first
2. A series of checks performed to verify if the candidate site is an actual Edge Delivery site
3. The flow checks both the site candidate and the sites tables to determine if Spacecat already has knowledge of the candidate. If yes, the candidate is ignored
4. An interactive message, containing [yes] and [ignore] buttons, is sent to a private Slack channel
5. If the human persona chooses to promote the site candidate to a site, it is then promoted accordingly
6. A site discovery message is sent to a public Slack channel

SpaceCat (Star Catalogue) - Site discovery

***

**Example interactive message sent to private channel for confirmation:**

Screenshot 2024-02-12 at 18 21 59

Screenshot 2024-02-12 at 18 35 51

***

**Example discovery announcement message sent to public channel:**

Screenshot 2024-02-12 at 18 35 59

### CODE REVIEW

File: src/controllers/hooks.js [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+const CDN_HOOK_SECRET_NAME = 'INCOMING_WEBHOOK_SECRET_CDN';
+const RUM_HOOK_SECRET_NAME = 'INCOMING_WEBHOOK_SECRET_RUM';
+
+export const BUTTON_LABELS = {
+  APPROVE_CUSTOMER: 'As Customer',
+  APPROVE_FRIENDS_FAMILY: 'As Friends/Family',
+  IGNORE: 'Ignore',
+};
+
+const IGNORED_SUBDOMAIN_TOKENS = ['demo', 'dev', 'stag', 'qa', '--'];
```
Comment (AndreiAlexandruParaschiv) [inline src/controllers/hooks.js:31]: stage or stg?

File: package.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
     "build": "hedy -v --test-bundle",
     "deploy": "hedy -v --deploy --test",
     "deploy-routes": "hedy --no-build -no-hints -l major",
-    "deploy-ci": "hedy -v --deploy --test --pkgVersion=ci$CIRCLE_BUILD_NUM -l ci --cleanup-ci=24h",
+    "deploy-ci": "hedy -v --deploy --test --pkgVersion=ci$CIRCLE_BUILD_NUM -l ci --cleanup-ci=48h",
```
Comment (AndreiAlexandruParaschiv) [inline package.json:18]: why the increase to 48h?

File: src/controllers/hooks.js (hunk 1 above)
Comment (ekremney) [reply src/controllers/hooks.js:31]: used `stag` to capture both `stage` and `staging` > both seem to exist in the logs

File: package.json (hunk 2 above)
Comment (ekremney) [reply package.json:18]: just for the testing on actual env. I will revert the changes to package.json before merging 👍

File: package.json [hunk 3]
```diff
@@ -29,7 +29,7 @@
     "memory": 4096,
     "awsRole!important": "arn:aws:iam::282898975672:role/spacecat-role-services",
     "testUrl": "/_status_check/healthcheck.json",
-    "awsAttachAuthorizer": "spacecat-token-authorizer",
+    "awsAttachAuthorizer": "",
```
Comment (solaris007) [inline package.json:32]: this should still use the authorizer...

File: src/controllers/hooks.js (hunk 1 above)
Comment (solaris007) [reply src/controllers/hooks.js:31]: _stag_: a male deer, especially a male red deer after its fifth year.

File: src/controllers/hooks.js [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+  if (!finalResp.ok) {
+    throw new InvalidSiteCandidate(`.plain.html does not return 2XX for ${finalUrl}`);
+  }
+
+  const respText = await finalResp.text();
+
+  // reject if .plain.html contains <head>
+  if (respText.includes('<head>')) {
+    throw new InvalidSiteCandidate('.plain.html should not contain <head>');
+  }
```
Comment (solaris007) [inline src/controllers/hooks.js:104]: could be shortened to:
```suggestion
   try {
     const finalResp = await fetch(finalUrl, { redirect: 'manual' });
     if (!finalResp.ok) {
       throw new InvalidSiteCandidate(`.plain.html does not return 2XX for ${finalUrl}`);
     }
     const respText = await finalResp.text();
     if (respText.includes('<head>')) {
       throw new InvalidSiteCandidate('.plain.html should not contain <head>');
     }
   } catch (e) {
     throw new InvalidSiteCandidate(`.plain.html is unreachable for ${finalUrl}`, { cause: e });
   }
```

File: src/support/slack/actions/commons.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        .text(blocks[0]?.text?.text),
+      Blocks.Section().text(reaction),
+    )
+    .buildToObject();
+
+  return {
+    ...message,
+    text: blocks[0]?.text?.text,
+    replace_original: true,
+  };
```
Comment (solaris007) [inline src/support/slack/actions/commons.js:56]: could be shortened to:
```suggestion
   const reaction = approved
     ? `Added by @${username} \`${hasText(orgId) ? BUTTON_LABELS.APPROVE_FRIENDS_FAMILY : BUTTON_LABELS.APPROVE_CUSTOMER}\` :checked:`
     : `Ignored by @${username} :cross-x:`;

   return {
     blocks: [
       { ...blocks[0] },
       Blocks.Section().text(reaction).build(),
     ],
     text: blocks[0]?.text?.text,
     replace_original: true,
   };
```

File: package.json (hunk 3 above)
Comment (ekremney) [reply package.json:32]: it's only there for testing, will revert it before merging 👍

File: src/controllers/hooks.js (hunk 4 above)
Comment (ekremney) [reply src/controllers/hooks.js:104]: moving the extra checks inside try blocks would result in inaccurate error messages such as:

```
.plain.html is unreachable for some-url.com
    .plain.html should not contain <head>
```

this log would be wrong when the some-url.com is reachable but also contains a `<head>`. I will keep the `!finalResp.ok` and `respText.includes('<head>')` checks out of the `try` block

**Comment by solaris007** [comment]
this PR seems to depend on https://github.com/adobe/spacecat-api-service/pull/137

**Pull request merged**